AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -505,7 +508,7 @@ - (AVAsset *)asset
 
 - (void)_setAndFetchAsset:(AVAsset *)asset url:(NSURL *)assetURL
 {
-  [self clearFetchedData];
+  [self _clearPlayer]; // Clear the player but not the underlying ASNetworkImageNode to avoid clearing the placeholder image and showing the background before the video starts.
   _asset = asset;
```

Comment:
@master-nevi could you describe a bit more the intended behavior here?

Should we only avoid clearing the placeholder in certain cases?  If you change the asset to a different one that has an unrelated contents, then it may be correct to clear the placeholder.

For example, should we check if URL (placeholder URL) is set?  And then, if it is NOT set, maybe clear the placeholder here?

I think the unit test should not need to call recursivelyClearFetchedData to get the right behavior, particularly if it is using the codepath where the asset's first frame is used as the thumbnail rather than a separate URL.